backend/mora/cli/__init__.py

```diff
@@ -6,5 +6,29 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 
+import os
+import sys
+
+# Can't do monkeypatching only because file is used as program
+try:
+    # first file patching in case we have test module installed:
+    import oio_common
+    from oio_common import db_structure
+    orgfile = db_structure.__file__
+    backenddir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+    topdir = os.path.dirname(backenddir)
+    patchfile = os.path.join(topdir, 'setup', "db_structure.py")
+    if not open(orgfile).read() == open(patchfile).read():
+        open(orgfile, "w").write(open(patchfile).read())
+        # monkeypatching necessary for
+        # first time running tests (jenkins)
```

Comment:
👎 — on general principle, test logic should reside in `tests`, and nowhere else. Furthermore, this file is not the proper location for this logic; it's just too magic to have this occur as a side-effect of importing `mora.cli` package.

When do we need to do this? If only during tests, we should do this in the `setUp` or `setUpClass` methods on the base class. (Not only will this make the tests more self-contained, it'll also ensure that the tests don't _rely_ on being called by our CLI.)

Reply:
I eneded up asking Jo, where she'd put it - I wanted to keep it out of setUpClass, as we don't inherit from the same class in all tests.
Jo suggested __init__.py in tests, and I went with it - it seems to avoid import during flask run, and that was, I guess, the main issue.

Reply:
> I eneded up asking Jo, where she'd put it - I wanted to keep it out of setUpClass, as we don't inherit from the same class in all tests.
> Jo suggested **init**.py in tests, and I went with it - it seems to avoid import during flask run, and that was, I guess, the main issue.

All tests that access LoRA descend from the same test class and/or mixin that grants them access to the database. To allow sharing it between LoRA and os2mo, the actual implementation currently exists in `oio_rest.utils.test_support` — but we have subclasses in `os2mo` where you can add an extra method.

I respectfully disagree with Io in this regard — having code of this order in `__init__.py` is simply too implicit and magical.